lib/libmbedtls/core/tee_lmd_provider.c

```diff
@@ -507,55 +510,94 @@ TEE_Result crypto_hash_final(void *ctx, uint32_t algo, uint8_t *digest,
  ******************************************************************************/
 
 #if defined(_CFG_CRYPTO_WITH_ACIPHER)
-struct bignum *crypto_bignum_allocate(size_t size_bits __unused)
+#define ciL		(sizeof(mbedtls_mpi_uint))	/* chars in limb  */
+#define biL		(ciL << 3)			/* bits  in limb  */
+#define biH		(ciL << 2)			/* half limb size */
+#define BITS_TO_LIMBS(i)		((i) / biL + ((i) % biL != 0))
+
+#define LMD_MAX_BITS_PER_VARIABLE	(4096)
+
+size_t crypto_bignum_num_bytes(struct bignum *a)
 {
-	return NULL;
+	assert(a != NULL);
+	return mbedtls_mpi_size((const mbedtls_mpi *)a);
 }
 
-TEE_Result crypto_bignum_bin2bn(const uint8_t *from __unused,
-				size_t fromsize __unused,
-				struct bignum *to __unused)
+size_t crypto_bignum_num_bits(struct bignum *a)
 {
-	return TEE_ERROR_NOT_IMPLEMENTED;
+	assert(a != NULL);
+	return mbedtls_mpi_bitlen((const mbedtls_mpi *)a);
 }
 
-size_t crypto_bignum_num_bytes(struct bignum *a __unused)
+int32_t crypto_bignum_compare(struct bignum *a, struct bignum *b)
 {
-	return 0;
+	int ret;
+
+	assert(a != NULL);
+	assert(b != NULL);
+	ret = mbedtls_mpi_cmp_mpi((const mbedtls_mpi *)a,
+		(const mbedtls_mpi *)b);
```

Comment:
Please align with the opening `(` above.

Reply:
Yes, I will fix it.